nri-kafka/src/Kafka/Worker/Fetcher.hs

```diff
@@ -11,7 +11,7 @@ import qualified Kafka.Worker.Partition as Partition
 import qualified Kafka.Worker.Settings as Settings
 import qualified Prelude
 
-type EnqueueRecord = (ConsumerRecord -> Prelude.IO Partition.SeekCmd)
+type EnqueueRecord = (ConsumerRecord -> Prelude.IO (Maybe Partition.SeekCmd))
```

Comment:
🎨 Instead of making this a `Maybe`, I propose we return a `Partition.NoSeek` instead of `Nothing`.

Reply:
I wasn't sure if that would cause something else?